examples/nodejs/nodejs.spec.ts

```diff
@@ -0,0 +1,31 @@
+import {mockRequest} from "../../src/index"
+import {fetchFromExampleDomain} from "./index"
+
+describe("Node.js example", () => {
+    it("should send a request to https://example.com", async () => {
+        // Set up your mockRequest:
+        const mockedRequest = mockRequest({
+            requestPattern: "https://example.com",
+            requestMethod: "GET",
+            responseStatus: 200,
+            responseBody: "Example mocked body",
+        })
+
+        // You can assert that the request has not been made in this test:
+        let expectedErrorWhenRequestNotMade
+        try {
+            mockedRequest.expectRequestToHaveBeenMade()
+        } catch (error) {
+            expectedErrorWhenRequestNotMade = error
+        }
+        expect(expectedErrorWhenRequestNotMade.message).toMatch(
+            "You expected that the request has been made, but it was not."
+        )
```

Comment:
Just an observation that this part of the API is a bit clunky, think we should be able to do this better with a `.toThrow(expectedError)` if we set it up better

Reply:
Ignore me, took this example from the lower level tests